source/common/http/conn_pool_base.cc

```diff
@@ -1,16 +1,66 @@
 #include "common/http/conn_pool_base.h"
 
+#include "common/network/transport_socket_options_impl.h"
+#include "common/runtime/runtime_features.h"
 #include "common/stats/timespan_impl.h"
 #include "common/upstream/upstream_impl.h"
 
 namespace Envoy {
 namespace Http {
+Network::TransportSocketOptionsSharedPtr
+wrapTransportSocketOptions(Network::TransportSocketOptionsSharedPtr transport_socket_options,
+                           Protocol protocol) {
+  if (!Runtime::runtimeFeatureEnabled("envoy.reloadable_features.http_default_alpn")) {
+    return transport_socket_options;
+  }
+
+  // TODO(reviewers): Given that we pass along the ALPN fallback over the same public interface that
+  // is provided as part of the LBContext, its possible that the ALPN fallback has already been set
+  // on the object. Right now we just stomp on it, which isn't great.
+  //
+  // This is further complicated by the fact that the TransportSocketOptions object is hashed to
+  // determine which conn pool to use: this is fine right now since we already include the protocol
+  // in the hash, but might set us up poorly for other changes that might want to follow the same
+  // pattern.
+  //
+  // One option would be to introduce a new parameter to createConnection so that the ALPN fallback
+  // can't be set on something that is exposed by the LB interface. It feels a bit odd to have two
+  // different parameters that both control the transport socket, but it would reduce the ambiguity
+  // of how this all works.
+
+  // If configured to do so, we override the ALPN to use for the upstream connection to match the
+  // selected protocol.
+  std::vector<std::string> alpn;
+  switch (protocol) {
+  case Http::Protocol::Http10:
```

Comment:
Per previous discussion I don't think this can ever happen? Either NOT_REACH this or just have this fall into the 1.1 case?